ipi/utils/io/backends/io_units.py

```diff
@@ -0,0 +1,59 @@
+"""Functions used to transform units in input files into default atomic system of units
+"""
+
+# This file is part of i-PI.
+# i-PI Copyright (C) 2014-2015 i-PI developers
+# See the "licenses" directory for full license information.
+
+
+import re
+from ipi.engine.atoms import Atoms
+from ipi.engine.cell import Cell
+from ipi.utils.units import unit_to_internal
+from ipi.engine.properties import Trajectories as Traj
+
+# Regular expressions initialization for read_xyz function
+cell_unit_re = re.compile('\s\{[a-z]*\}\s')             # cell unit pattern
+traj_dict = Traj().traj_dict                            # trajectory dictionary
+traj_re = [re.compile('%s%s' % (key, '\{[a-z]*\}')) for key in traj_dict.keys()]  # trajectory patterns
+
+def process_units(comment, cell, qatoms, names, masses, output='objects'):
+
+  # Extracting trajectory units
+  family, unit = 'undefined', ''
+  if comment != '':
+    traj = filter(None, [key.search(comment) for key in traj_re])[0].group()[:-1].split('{')
+    family, unit = traj_dict[traj[0]]['dimension'], traj[1]
```

Comment:
**WARNING:**

If comment do not contains any matchable string, the program will enter the `if` but `traj` will be an empty list  raising an error in the line below.

Replacing lines 24-25 with the following should fix it.

```python
  is_comment_useful = filter(None, [key.search(comment.strip()) for key in traj_re])
  if len(is_comment_useful) > 0:
    traj = is_comment_useful[0].group()[:-1].split('{')
```